tests/test_matplotlibFormatter.py

```diff
@@ -0,0 +1,73 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Tests for MatplotlibFormatter.
+"""
+
+import unittest
+import tempfile
+import os
+import shutil
+
+import matplotlib
+matplotlib.use('Agg')  # noqa:E402
+from matplotlib import pyplot
+import numpy as np
+from lsst.daf.butler import Butler, DatasetType
+import filecmp
+
+TESTDIR = os.path.abspath(os.path.dirname(__file__))
+
+
+class MatplotlibFormatterTestCase(unittest.TestCase):
+    """Test for MatplotlibFormatter.
+    """
+
+    def setUp(self):
+        self.root = tempfile.mkdtemp(dir=TESTDIR)
+        Butler.makeRepo(self.root)
+
+    def tearDown(self):
+        if os.path.exists(self.root):
+            shutil.rmtree(self.root, ignore_errors=True)
+
+    def testMatplotlibFormatter(self):
+        butler = Butler(self.root, run='testrun')
+        datasetType = DatasetType('test_plot', [], 'Plot',
+                                  universe=butler.registry.dimensions)
+        butler.registry.registerDatasetType(datasetType)
+        pyplot.imshow(np.random.randn(3, 4))
+        butler.put(pyplot.gcf(), datasetType)
+        with tempfile.NamedTemporaryFile(suffix='.png') as file:
+            pyplot.gcf().savefig(file.name)
+            # predicting the filename path based on test run
+            self.assertTrue(
+                filecmp.cmp(
+                    os.path.join(self.root, 'testrun', datasetType.name,
```

Comment:
I think I would prefer it if this path was constructed from calling `butler.getUri` since then you aren't relying on knowing how the file name was constructed within the datastore.

Reply:
That's what we tried first, but annoyingly `filecmp` apparently can't handle `file://` prefixes, and it occurred to me that *actual* usage would look more like this, but perhaps with users explicitly defining the template so they'd know where to look in general, rather than asking the butler where to look each time.

Do you have an incantation handy for how to get that `file://` prefix off gracefully?

Reply:
You can either use `ButlerURI` or you can go for the standard option of:
```python
parsed = urllib.parse.urlparse(uri)
```
and then use `parsed.path`.